controllers/btpoperator_controller_test.go

```diff
@@ -224,29 +221,27 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
 			deleteSecret := &corev1.Secret{}
 			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: kymaNamespace, Name: SecretName}, deleteSecret)).To(Succeed())
 			Expect(k8sClient.Delete(ctx, deleteSecret)).To(Succeed())
-			k8sClient.Get(ctx, client.ObjectKey{Namespace: defaultNamespace, Name: btpOperatorName}, cr)
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: defaultNamespace, Name: btpOperatorName}, cr)).To(Succeed())
 			cr.SetLabels(map[string]string{forceDeleteLabelKey: "true"})
-			k8sClient.Update(ctx, cr)
-			time.Sleep(time.Second)
+			Expect(k8sClient.Update(ctx, cr)).To(Succeed())
+			Eventually(func() (bool, error) {
+				err := k8sClient.Get(ctx, client.ObjectKey{Namespace: defaultNamespace, Name: btpOperatorName}, cr)
+				return cr.Labels[forceDeleteLabelKey] == "true", err
+			}).WithTimeout(k8sOpsTimeout).WithPolling(k8sOpsPollingInterval).Should(BeTrue())
+			Eventually(updateCh).Should(Receive(matchDeleted()))
 		})
 
 		It("Delete should fail because of existing instances and bindings", func() {
-			siUnstructured = createResource(instanceGvk, kymaNamespace, instanceName)
+			_ = createResource(instanceGvk, kymaNamespace, instanceName)
 			ensureResourceExists(instanceGvk)
 
-			sbUnstructured = createResource(bindingGvk, kymaNamespace, bindingName)
+			_ = createResource(bindingGvk, kymaNamespace, bindingName)
 			ensureResourceExists(bindingGvk)
 
-			reconciler.Client = newTimeoutK8sClient(reconciler.Client)
-			setFinalizers(siUnstructured)
```

Comment:
why did you remove setting finalizers?

Reply:
the test seems ok without finalizers, deprovisioning will fail because of existing SI/SB and lack of `force-delete` label. Adding finalizers only change how BtpOperatorReconciler proceeds when it's able to deprovision remaining resources